README.md

```diff
@@ -84,11 +85,59 @@ Since the alignment may jump to a previous reference position, different segment
 
 The output is in SAM/BAM format.
 If the alignment has `N` jumps, then the output will contain `N+1` records for the input read.
-One record is marked as primary (see `--pick-primary`), while the remaining records are marked as secondary.
-The HI/HN SAM tags are used to denote the order in which the alignments occur.
-Furthermore, the records are output by this tool in the order in which they align to the query/read sequence.
-
-Multiple contigs in the input FASTA are supported.
+In some cases, multiple "chains" of alignments are output.
+See [SAM Flags and Tags](#sam-flags-and-tags) for details on how the SAM flags are set as well as
+custom SAM tags, to reason which alignments are part of which linear alignment chain, and which
+linear alignment chain is the "primary".
+
+Multiple contigs in the input FASTA are supported (and encouraged).
+
+### SAM Flags and Tags
+
+In the simplest case, a read (query) aligns to on contig (target) sub-sequence, such that a single
+SAM alignment record is output.
+If a read has a linear alignment (chain), whereby (typically) non-overlapping subsequence of the
+read are aligned to various sub-sequences of the reference (may be different contigs), then one
+SAM alignment record is output per sub-alignment in the chain.
+In some cases, there may be multiple chains of alignments reported, such that one linear chain
+is the "representative" (or best) linear alignment of the read, and the others are "secondary".
+In these cases, SAM alignment records are output for each sub-alignment across each chain.
+
+For example, there may be three (3) linear alignments of the read, with the first having six (6)
+sub-alignments (i.e. five (5) jumps), the second having two (2) sub-alignments (i.e. one jump),
+and the third having a single sub-alignment (i.e. no jumps).
+In this case, there will be nine (9) SAM alignment records output (6 + 2 + 1).
+
+SAM flags and custom SAM tags are used to identify the chain the SAM alignemnt record belongs, as 
+well as the order of the alignment records in the given linear chain.  In particular, SAM flags
+are set as follows:
+1. One sub-alignment will (a) not have the secondary flag set, and (b) not have the
+   supplementary flag set.  This is the "primary" (representative) sub-alignment in the
+   "primary chain" (NB: not a formal term).
+2. The remaining sub-alignments in the "primary chain" have the supplementary flag set,
+   but not the secondary flag set.  They are part of the best linear alignment chain,
+   but not the representative sub-alignment.
+3. For "secondary chains", have one sub-alignment be representative.  Set the secondary
+   flag (since it's not in the "primary chain") but do not set the supplementary flag.
+4. For "secondary chains", have the non-representative sub-alignments have both the
+   secondary and supplementary flags set.
+
+See `--pick-primary` for how the primary alignment is chosen (and so when the secondary flag is 
```

Comment:
Not sure what exactly you mean in the parenthetical statement - consider revising

Reply:
Just removed as it is confusing, and duplicates the description above!